src/lava/magma/compiler/channels/pypychannel.py

```diff
@@ -265,6 +273,43 @@ def join(self):
         self._done = True
 
 
+class CspSelector:
+    """
+    Utility class to allow waiting for multiple channels to become ready
+    """
+
+    def __init__(self):
+        """Instantiates CspSelector object and class attributes"""
+        self._cv = Condition()
+
+    def _changed(self, channel):
+        with self._cv:
+            self._cv.notify_all()
+
+    def _set_observer(self, channel_actions, observer):
+        for channel, _ in channel_actions:
+            channel.observer = observer
+
+    def select(
+        self,
+        *args: ty.Tuple[ty.Union[CspSendPort, CspRecvPort],
+                        ty.Callable[[], ty.Any]
+                        ]
+    ):
+        """
+        Wait for any channel to become ready, then execute the corresponding
+        callable and return the result.
+        """
+        with self._cv:
+            self._set_observer(args, self._changed)
```

Comment:
Why do we have to set and unset this all the time?

Reply:
In an earlier version, all channels notified, and _changed() determined if the source channel is part of the current set being selected on.  (This is also the reason for the now unused channel argument, which I didn't bother to remove.)  I thought doing it this way was cheaper and cleaner.